OPTIM use pairwise_distances_argmin in NearestCentroid.predict (#24645)
scikit-learn/scikit-learn: doc/whats_new/v1.2.rst

  :pr:`10468` by :user:`Ruben <icfly2>` and :pr:`22993` by
  :user:`Jovan Stojanovic <jovan-stojanovic>`.

> - |Efficiency| :class:`neighbors.NearestCentroid` requires less memory and better
>   leverages the CPU cache to compute predictions.
>   :pr:`24645` by :user:`ogrisel`.

Reviewer: Nit.
```suggestion
- |Efficiency| :class:`neighbors.NearestCentroid` is faster and requires
  less memory as it better leverages CPUs' caches to compute predictions.
  :pr:`24645` by :user:`Olivier Grisel <ogrisel>`.
```
